tests/unittests/GraphOptzTest.cpp

```diff
@@ -401,29 +401,40 @@ TEST_F(GraphOptz, cseRespectsPredicates) {
 }
 
 TEST_F(GraphOptz, optimizeBatchNormAfterConvButConvReused) {
-  Node *A =
+  Placeholder *A =
       mod_.createPlaceholder(ElemKind::FloatTy, {1, 10, 20, 3}, "A", false);
   Node *CV = F_->createConv(bindings_, "conv", A, 16, 5, 1, 2, 1);
   Node *BN =
       F_->createBatchNormalization(bindings_, "batch", CV, 3, 0.0001, 0.9);
-  SaveNode *ret = F_->createSave("ret", BN);
-  SaveNode *convSave = F_->createSave("convSave", CV);
+  F_->createSave("ret", BN);
+  F_->createSave("convSave", CV);
 
   EXPECT_EQ(F_->getNodes().size(), 4);
-
-  ::glow::optimize(F_, CompilationMode::Infer);
+  optimizedF_ = optimizeFunction(F_);
   // Make sure the structure of the graph did not change, since the convolution
   // node is used more than once.
-  EXPECT_EQ(F_->getNodes().size(), 4);
-  EXPECT_TRUE(llvm::isa<ConvolutionNode>(convSave->getInput()));
-  ConvolutionNode *conv = llvm::dyn_cast<ConvolutionNode>(convSave->getInput());
-  EXPECT_EQ(conv, CV);
-  EXPECT_TRUE(llvm::isa<BatchNormalizationNode>(ret->getInput()));
-  BatchNormalizationNode *batchNorm =
-      llvm::dyn_cast<BatchNormalizationNode>(ret->getInput());
-  EXPECT_EQ(batchNorm, BN);
-  EXPECT_EQ(batchNorm->getInput().getNode(), CV);
+  EXPECT_EQ(optimizedF_->getNodes().size(), 4);
+  ConvolutionNode *conv = nullptr;
+  BatchNormalizationNode *batchNorm = nullptr;
+  for (auto &N : optimizedF_->getNodes()) {
+    if (N.getKind() == Kinded::Kind::SaveNodeKind) {
+      SaveNode *temp = llvm::dyn_cast<SaveNode>(&N);
+      if (llvm::isa<ConvolutionNode>(temp->getInput()) == true) {
+        conv = llvm::dyn_cast<ConvolutionNode>(temp->getInput());
+      } else {
+        batchNorm = llvm::dyn_cast<BatchNormalizationNode>(temp->getInput());
+      }
+    }
+  }
```

Comment:
Consider shortening this using `std::find/std::find_if/std::find_if_not`: https://en.cppreference.com/w/cpp/algorithm/find